### ISSUE TITLE
Abort test suite execution from before_*/after_* hooks?

### ISSUE THREAD

**Issue by joshsleeper** [behave/behave#475]
We have some conditions we can detect at the before_* or after_* hook level where we'd like to totally and immediately abort the current test suite.

For example, if some mandatory cleanup in our `before_scenario()` logic fails, we'd like to print out some error text, abort the test suite, and then have Behave print the standard pass/fail/skip information for whatever tests it did run.

Right now the only options I've found to do something like this are:

1. Raise an error that Behave doesn't catch (although I don't believe it catches much from those hooks, so most errors work), which _works_ for aborting the test suite but prevents us from getting final output for what tests it ran.
2. Set some global flag such that all successive tests are forced to skip from that point on.

Both of those are doable, but neither is ideal by any means.
Is there any way to abort test suite execution like this as of 1.2.5?

**Comment by jenisys** [comment]
Look at [features/runner.abort_by_user.feature](https://github.com/behave/behave/blob/master/features/runner.abort_by_user.feature). In general, you should use an exception that is derived ``BaseException`` for that. ``KeyboardInterrupt``, ``SystemExit`` (or derived) should be handled gracefully.

A more graceful method would use ``runner.aborted = True``, but that is normally not accessible to a user (by intention). You may try something, like the following:

```python
# -- FILE: features/environment.py
# HACK: Not part of the official API

def before_scenario(context, scenario):
    """Code applies to feature part like:

    .. code-block:: gherkin

        @abort_in_hook.before_scenario
        Scenario: Expected to abort here
           Given ...
    """
    if "abort_in_hook.before_scenario" in scenario.tags:
        context._runner.aborted = True
```

**Comment by joshsleeper** [comment]
Well how about that.

`KeyboardInterrupt` did exactly what I wanted, thanks so much @jenisys!

Also, just fyi, I tried using `context._runner.aborted = True` and couldn't get it to work properly.
It _would_ print out that the test run was aborted, but not until _after_ running all of the tests.
